### PR TITLE
fix(auth): make createUser idempotent and cap passwords at 72 chars

### PR DESCRIPTION
**Pull request by DonKoko**
Two signup failures from the Sentry triage.

createUser (SHELF-WEBAPP-1EA): a P2002 unique-constraint violation on the User primary key (id = the Supabase auth id) threw "We had trouble while creating your account" — hit on re-signup or a prior partial signup whose stored email differs from the OTP email (so the route's email-keyed race guard misses it). Since user.create and all its side-effects run in one $transaction, the P2002 rolls the whole thing back with no partial state, so the catch now returns the pre-existing row (identical select shape) instead of throwing. The signup_completed event is not re-fired; a null lookup (P2002 on another unique field) still throws.

Password length (SHELF-WEBAPP-21A): the signup/onboarding/reset flows had no max length, so a >72-char password surfaced Supabase's bcrypt limit (AuthApiError) as a generic captured 500. Add a shared passwordSchema(minMsg?) factory in utils/zod.ts (min 8, max 72) and apply it to join, onboarding (setter branch only — the optional branch stays unconstrained), and forgot-password, preserving each flow's existing min-length copy. Add the required server-side validation-error fallback to the join password field.

Fixes SHELF-WEBAPP-1EA
Fixes SHELF-WEBAPP-21A

## Summary by CodeRabbit

* **New Features**
  * Added shared password validation across join, onboarding, and forgot-password flows, including consistent max-length enforcement and unified “too short” messaging.
  * Improved password inputs to display field-specific server validation errors.

* **Bug Fixes**
  * Prevented duplicate account creation during signup races by reusing the existing account when unique-constraint conflicts occur.
  * Ensured recovered signups don’t re-trigger signup-completion tracking, and reliably re-attach the user to the requested organization when needed.

* **Tests**
  * Expanded coverage for password schema edge cases and idempotent account creation behavior.

### CODE REVIEW

File: apps/webapp/app/modules/user/service.server.ts [hunk 1]
```diff
@@ -848,6 +848,40 @@ export async function createUser(
...
+          organizations: {
+            select: {
+              id: true,
+            },
+          },
+        },
+      });
+
+      if (existingUser) {
+        return existingUser;
```
[the remark this answers is not in the document]
Comment (DonKoko) [reply apps/webapp/app/modules/user/service.server.ts:906]: Fixed in `38feae536`. The idempotent `createUser` P2002 path now **reconciles the org association**: when `organizationId` is present (invite/SSO callers) and the existing user isn't already a member, it attaches them via `createUserOrgAssociation`. It's idempotent — the `!existingUser.organizations.some(...)` membership check skips re-attaching (so the upsert's `roles: { push }` branch can't duplicate roles). The personal-org/OTP case has no `organizationId`, so nothing to reconcile there. Added tests for both attach-when-missing and skip-when-already-a-member.

File: apps/webapp/app/utils/zod.ts [hunk 2]
```diff
@@ -35,3 +35,37 @@ export const stringToJSONSchema = z
...
+ *   existing wording. Defaults to the signup message.
+ * @returns A `z.ZodString` schema with `min(8)` and `max(72)` checks.
+ */
+export function passwordSchema(
+  minMessage = "Your password is too short. Min 8 characters are required."
+) {
+  return z
+    .string()
+    .min(8, minMessage)
+    .max(PASSWORD_MAX_LENGTH, PASSWORD_MAX_LENGTH_MESSAGE);
```
[the remark this answers is not in the document]
Comment (DonKoko) [reply apps/webapp/app/utils/zod.ts]: Fixed in `38feae536`. `passwordSchema` now refines on **UTF-8 byte length** — `new TextEncoder().encode(value).length <= 72` — instead of `.max(72)` (which counts UTF-16 code units). So a multi-byte password like `"é".repeat(40)` (40 chars / 80 bytes) is now correctly rejected before it reaches Supabase. Added multi-byte test cases (`"é".repeat(37)` = 74 bytes rejected, `"é".repeat(36)` = 72 bytes accepted).

File: apps/webapp/app/modules/user/service.server.ts [hunk 3]
```diff
@@ -848,6 +848,40 @@ export async function createUser(
...
+      }
```
[the remark this answers is not in the document]
Comment (DonKoko) [reply apps/webapp/app/modules/user/service.server.ts:907]: Fixed in `38feae536`. The idempotent `createUser` P2002 path now **reconciles the org association**: when `organizationId` is present (invite/SSO callers) and the existing user isn't already a member, it attaches them via `createUserOrgAssociation`. It's idempotent — the `!existingUser.organizations.some(...)` membership check skips re-attaching (so the upsert's `roles: { push }` branch can't duplicate roles). The personal-org/OTP case has no `organizationId`, so nothing to reconcile there. Added tests for both attach-when-missing and skip-when-already-a-member.

File: apps/webapp/app/routes/_auth+/join.tsx [hunk 4]
```diff
@@ -182,7 +180,10 @@ export default function Join() {
             autoComplete="new-password"
             disabled={disabled}
             inputClassName="w-full"
-            error={zo.errors.password()?.message}
+            error={
+              getValidationErrors<typeof JoinFormSchema>(data?.error)?.password
+                ?.message || zo.errors.password()?.message
+            }
```
[the remark this answers is not in the document]
Comment (DonKoko) [reply apps/webapp/app/routes/_auth+/join.tsx:186]: Fixed in `38feae536`. Added the `getValidationErrors` server-side fallback to **every** password input per the CLAUDE.md form pattern: join's `confirmPassword`, both onboarding password fields, and both forgot-password fields. For forgot-password specifically, a validation error now renders inline on the fields instead of bouncing back to the email step — while a hard error (e.g. an invalid OTP) still falls back to the generic message.

File: apps/webapp/app/utils/zod.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (DonKoko) [reply apps/webapp/app/utils/zod.ts]: Fixed in `38feae536`. `passwordSchema` now refines on **UTF-8 byte length** — `new TextEncoder().encode(value).length <= 72` — instead of `.max(72)` (which counts UTF-16 code units). So a multi-byte password like `"é".repeat(40)` (40 chars / 80 bytes) is now correctly rejected before it reaches Supabase. Added multi-byte test cases (`"é".repeat(37)` = 74 bytes rejected, `"é".repeat(36)` = 72 bytes accepted).
